### PR TITLE
feat: Support and tests for person batch exports in Postgres

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

PR Adds support for person batch exports for PostgreSQL and Redshift. Works the same as its Bigquery (https://github.com/PostHog/posthog/pull/23187) and Snowflake (https://github.com/PostHog/posthog/pull/23333) equivalents.

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

Updated tests.

### CODE REVIEW

File: posthog/temporal/batch_exports/redshift_batch_export.py [hunk 1]
```diff
@@ -78,22 +79,104 @@ def remove_escaped_whitespace_recursive(value):
...
+        DELETE FROM {stage_table}
+        USING {final_table} AS final
+        WHERE {merge_condition} AND {stage_table}.{stage_version_key} > final.{final_version_key};
+        """).format(
+            final_table=final_table_identifier,
+            stage_table=stage_table_identifier,
+            merge_condition=delete_condition,
+            stage_version_key=sql.Identifier(version_key),
+            final_version_key=sql.Identifier(version_key),
+        )
```
Comment (tomasfarias) [inline posthog/temporal/batch_exports/redshift_batch_export.py:163]: Redshift doesn't support adding a condition on the merge, so we have to delete any existing rows here

Comment (tomasfarias) [reply posthog/temporal/batch_exports/redshift_batch_export.py:163]: Also, redshift doesn't support aliasing tables sometimes, it's very annoying.

File: posthog/temporal/batch_exports/postgres_batch_export.py [hunk 2]
```diff
@@ -35,120 +35,309 @@
...
+    apeek_first_and_rewind,
+    cast_record_batch_json_columns,
+    set_status_to_running_task,
 )
-from posthog.temporal.batch_exports.utils import apeek_first_and_rewind, set_status_to_running_task
 from posthog.temporal.common.clickhouse import get_client
 from posthog.temporal.common.heartbeat import Heartbeater
 from posthog.temporal.common.logger import bind_temporal_worker_logger
 
+PostgreSQLField = tuple[str, typing.LiteralString]
```
Comment (tomasfarias) [inline posthog/temporal/batch_exports/postgres_batch_export.py:49]: Yay Python 3.11, we now have `typing.LiteralString` and can get rid of a type: ignore comment.

**Review by Gilbert09 — APPROVED** [review]
Don't have a lot of context on exports, but both the postgres and redshift code looks good. Would probably be useful to let people use a SSH tunnel to connect to their instances though

**Comment by tomasfarias** [comment]
> Would probably be useful to let people use a SSH tunnel to connect to their instances though

Absolutely. I think it's out of scope of this PR, but I will put it on the roadmap. This is a feature I would love to support (perhaps by borrowing from data warehouse imports).

**Pull request merged**